keras_nlp/tokenizers/sentence_piece_tokenizer_test.py

```diff
@@ -0,0 +1,195 @@
+        input_data = ["the quick brown fox."]
+        tokenizer = SentencePieceTokenizer(
+            model_bytes=self.model_bytes,
+        )
+        call_output = tokenizer(input_data)
+        tokenize_output = tokenizer.tokenize(input_data)
+        self.assertIsInstance(call_output, tf.RaggedTensor)
+        self.assertAllEqual(call_output, [[6, 5, 3, 4]])
+        self.assertAllEqual(tokenize_output, [[6, 5, 3, 4]])
+
+    def test_scalar_tokenize(self):
+        input_data = "the quick brown fox."
+        tokenizer = SentencePieceTokenizer(
+            model_bytes=self.model_bytes,
+        )
+        call_output = tokenizer(input_data)
+        tokenize_output = tokenizer.tokenize(input_data)
+        self.assertIsInstance(call_output, tf.Tensor)
+        self.assertAllEqual(call_output, [6, 5, 3, 4])
+        self.assertAllEqual(tokenize_output, [6, 5, 3, 4])
+
+    def test_dense_output(self):
+        input_data = ["the quick brown fox."]
+        tokenizer = SentencePieceTokenizer(
+            model_bytes=self.model_bytes,
+            sequence_length=10,
+        )
+        output_data = tokenizer(input_data)
+        self.assertIsInstance(output_data, tf.Tensor)
+        self.assertAllEqual(output_data, [[6, 5, 3, 4, 0, 0, 0, 0, 0, 0]])
+
+    def test_string_tokenize(self):
+        input_data = ["the quick brown fox."]
+        tokenizer = SentencePieceTokenizer(
+            model_bytes=self.model_bytes,
+            dtype="string",
+        )
+        output_data = tokenizer(input_data)
+        self.assertAllEqual(
+            output_data,
+            tf.ragged.constant([["▁the", "▁quick", "▁brown", "▁fox."]]),
+        )
+
+    def test_detokenize(self):
+        input_data = [[6, 5, 3, 4]]
+        tokenizer = SentencePieceTokenizer(
+            model_bytes=self.model_bytes,
+        )
+        output_data = tokenizer.detokenize(input_data)
+        self.assertAllEqual(output_data, ["the quick brown fox."])
+
+    def test_accessors(self):
+        tokenizer = SentencePieceTokenizer(
+            model_bytes=self.model_bytes,
+        )
+        self.assertEqual(tokenizer.vocabulary_size(), 7)
+        self.assertEqual(tokenizer.id_to_token(0), "<unk>")
+        self.assertEqual(tokenizer.id_to_token(5), "▁quick")
+        self.assertEqual(tokenizer.token_to_id("<unk>"), 0)
+        self.assertEqual(tokenizer.token_to_id("▁quick"), 5)
+
+    def test_functional_model(self):
+        input_data = tf.constant(["the quick brown fox."])
+        tokenizer = SentencePieceTokenizer(
+            model_bytes=self.model_bytes,
+        )
+        inputs = keras.Input(dtype="string", shape=())
+        outputs = tokenizer.detokenize(tokenizer.tokenize(inputs))
+        model = keras.Model(inputs, outputs)
+        model_output = model(input_data)
+        self.assertAllEqual(model_output, ["the quick brown fox."])
+
+    def test_from_file(self):
+        model_file = os.path.join(self.get_temp_dir(), "model.txt")
```

Comment:
where is this model.txt from?

Reply:
I write the file two lines below inside the test.